### PR TITLE
Fix reconnect

### PR DESCRIPTION
**Pull request by ekoby**
the chain of links (tcp -> tls -> http) gets taken apart when connection is closed. Fix to re-establish the chain when connection is established

### CODE REVIEW

File: .idea/fileTemplates/includes/C File Header.h [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #if ($HEADER_COMMENTS)
 /*
-Copyright 2019 NetFoundry, Inc.
+Copyright $YEAR NetFoundry, Inc.
```
[the remark this answers is not in the document]
Comment (dovholuknf) [reply .idea/fileTemplates/includes/C File Header.h]: actually - should change it to 2019-$YEAR. nevermind, that doesn't cover 'new files'... ugh. oh well.

File: sample/repeat-fetch.c [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+#include "common.h"
+
+int count = 5;
+um_http_t time_clt;
+uv_timer_t time_timer;
+
+static void timer_cb(uv_timer_t *timer) {
+    if (count-- > 0) {
+        printf(">>> calling time service count left = %d\n\n", count);
+        um_http_req_t *req = um_http_req(&time_clt, "GET", "/api/timezone/EST");
```
Comment (dovholuknf) [inline sample/repeat-fetch.c]: would it be nice to declare or find/reuse constants for stuff like "GET", "POST" etc?

File: src/http.c [hunk 3]
```diff
@@ -190,29 +190,48 @@ static void requests_fail(um_http_t *c, int code) {
...
-        uv_link_read_start(&c->http_link);
+        uv_link_source_init(&clt->conn_src, (uv_stream_t *) &clt->conn);
+        uv_link_init(&clt->http_link, &http_methods);
+        clt->http_link.data = clt;
 
-        if (!c->ssl) {
-            c->connected = true;
-            uv_async_send(&c->proc);
+        if (clt->ssl) {
+            clt->tls_link = malloc(sizeof(uv_link_t));
```
Comment (dovholuknf) [inline src/http.c]: does this ever get free'd? i see it in here getting allocated but scanning the PR i don't see it being free'd. maybe it's on a previous pr but figured i'd ask

File: .idea/fileTemplates/includes/C File Header.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (ekoby) [reply .idea/fileTemplates/includes/C File Header.h]: This is just for new files

File: src/http.c (hunk 3 above)
Comment (ekoby) [reply src/http.c]: changed it into a member struct (not pointer), no longer needs to be allocated/freed

**Pull request merged**